common/mas_knowledge_utils/tbox_yaml_loader.py

```diff
@@ -0,0 +1,199 @@
+                       if the ontology has to be exported to a new file. 
+                       If it is 'None', the loaded ontology will be updated with the definitions
+
+        '''
+        if export_path is None:
+            # Overwrite the existing ontology file with the updated ontology
+            self.__verbose("No export filepath specified. Updating the loaded ontology.")
+            self.ontology_if.update()
+        else:
+            # Save the updated ontology at the given file location
+            self.__verbose("Exporting the ontology to {0}".format(export_path))
+            self.ontology_if.export(export_path)
+
+    def __get_independent_classes(self):
+        '''Returns a list of class_names in the loaded class_definitions whose 
+        parent classes have already been defined in the ontology
+        '''
+        independent_classes = []
+        for name, parents in self.class_definitions.items():
+            is_independent = True
+            for parent in parents:
+                if not self.ontology_if.is_class(parent):
+                    is_independent = False
+                    break
+            if is_independent:
+                independent_classes.append(name)
+        if len(independent_classes) == 0 and len(self.class_definitions):
+            raise Exception('Could not find indpendent class names from the loaded class descriptions!')
+        return independent_classes
+
+    def __load_definitions(self, definitions_file):
+        '''Loads the class and property definitions as two separate
+        dictionaries from the yaml file
+
+        Keyword arguments:
+        definitions_file -- absolute-path to the yaml file containing definitions
+
+        '''
+        definitions = None
+        with open(definitions_file, 'r') as f:
+            definitions = yaml.load(f, Loader=yaml.FullLoader)
+
+        if definitions is None:
+            raise Exception("Could not load the definition file!")
+        else:
+            self.class_definitions = definitions['class_definitions']
+            self.property_definitions = definitions['property_definitions']
+
+            # Convert any loaded 'None' strings to None data-type
+            for property_name in self.property_definitions.keys():
+                domain = []
+                for d in self.property_definitions[property_name]['domain']:
+                    domain.append(self.__check_none(d))
+                self.property_definitions[property_name]['domain'] = domain
+
+                range = []
+                for r in self.property_definitions[property_name]['range']:
+                    range.append(self.__check_none(r))
+                self.property_definitions[property_name]['range'] = range
+
+                self.property_definitions[property_name]['type'] = \
+                    self.__check_none(self.property_definitions[property_name]['type'])
+    
+    def __check_none(self, value):
```

Comment:
Can you rename this to `__maybe_none`? That's a more correct way - in terms of functional programming - of referring to what you're trying to achieve here.

Reply:
Done